### PR TITLE
sanitize and escape HTML

### LINKED ISSUE #122: Keyword can contain HTML-tags
It look likes the keyword can contain HTML-tags:
add_new_post_ _theme_ _wordpress

Another effect of html are the `<i>` tags:
add_new_post_ _theme_ _wordpress

It seems the HTML will be parsed when the tags are started and closed.
add_new_post_ _theme_ _wordpress

### PR DESCRIPTION
**Pull request by terw-dan**
Fixes #122 

Removes HTML-tags from the keyword, so no HTML is analyzed and returned on the page
Sanitizes the output of the scoretexts

### CODE REVIEW

File: js/stringhelper.js [hunk 1]
```diff
@@ -129,10 +129,28 @@ YoastSEO.StringHelper.prototype.stripAllTags = function( textString ) {
...
 
+	keyword = this.stripAllTags( keyword );
+
 	return keyword;
 };
 
 /**
+ *
+ * @param textString
+ * @returns {*}
```
Comment (atimmer) [inline js/stringhelper.js]: We return a string not a star

File: js/stringhelper.js [hunk 2]
```diff
@@ -129,10 +129,28 @@ YoastSEO.StringHelper.prototype.stripAllTags = function( textString ) {
...
 	keyword = keyword.replace( /[\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "" );
 
+	keyword = this.stripAllTags( keyword );
+
 	return keyword;
 };
 
 /**
+ *
+ * @param textString
```
Comment (atimmer) [inline js/stringhelper.js]: We require a string, so it should be:
` * @param {string} textString`

File: js/stringhelper.js [hunk 3]
```diff
@@ -129,10 +129,28 @@ YoastSEO.StringHelper.prototype.stripAllTags = function( textString ) {
 YoastSEO.StringHelper.prototype.sanitizeKeyword = function( keyword ) {
 	keyword = keyword.replace( /[\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "" );
 
+	keyword = this.stripAllTags( keyword );
+
 	return keyword;
 };
 
 /**
+ *
```
Comment (atimmer) [inline js/stringhelper.js]: Please add function description.

File: js/app.js [hunk 4]
```diff
@@ -369,6 +369,8 @@ YoastSEO.App.prototype.runAnalyzer = function() {
 		this.analyzerData.queue = [ "keyWordCheck", "wordCount", "fleschReading", "pageTitleLength", "urlStopwords", "metaDescriptionLength" ];
 	}
 
+	this.analyzerData.keyword = keyword;
```
Comment (atimmer) [inline js/app.js]: What exactly does this achieve?

**Comment by atimmer** [comment]
We needs one test that tests what happens if you put tags in the keyword.

**Comment by atimmer** [comment]
CR Done 👍🏻

File: js/app.js (hunk 4 above)
Comment (terw-dan) [inline js/app.js]: This sends the keyword to the analyzer that is sanitized and stripped from HTML

**Comment by terw-dan** [comment]
processed feedback

**Pull request merged**